### DISCUSSION TITLE
Datagrid : allow default summary text to be hidden

### DISCUSSION THREAD

**Discussion by MBNSoftware** [microsoft/fluentui-blazor#1670] [Ideas]
Hello,

If using Paginator on a data grid, it always shows both summary and pagination informations, even if one of them is not provided.

This is because of the code in FluentPaginator.razor that is forcing a default value if the template is null (not provided).

Excerpt for the SummaryTemplate :

```
<div class="summary">
            @if (SummaryTemplate is not null)
            {
                @SummaryTemplate
            }
            else
            {
                <text><strong>@State.TotalItemCount</strong> items</text>
            }
        </div>
```

To me, a simple "if" statement would be enough : if you provide the template then it is displayed otherwise ... there is no otherwise :)

```
<div class="summary">
            @if (SummaryTemplate is not null)
            {
                @SummaryTemplate
            }
        </div>
```

By the way, the default text being hardcoded in english does not seem to be a good idea either, to me.

One side effect of removing the default text for the SummaryTemplate is that the PaginationTemplate is now aligned on the left.
While understandable (and good for my own needs), it may not be desirable for everybody.

![image](https://github.com/microsoft/fluentui-blazor/assets/70918793/3599eb93-2470-4166-9630-6ba6efbdc09e)

I don't know if a single "if" statement is the best way or if a parameter would be better. Sure enough, the if statement would be the easiest one to implement.

What do you think ?

A workaround (dirty workaround) is currently to provide a space in the template. This way, it is not null and may not be noticeable in the display.

Regards,
Christophe

**Comment by vnbaaij** [comment]
Tha Datagrid's design is to always have the summary and the pagination components. The defaults are there to make it easier to have something that 'just works'. 
You can change these parts to your liking by using the template parameters. If you do not want one of them, using an 'empty' template is the way to go right now. Don't see that as a hack and don't think we need to provide extra code or parameters for that.
Besides, you can always use CSS to hide and/or position elements.

**Comment by MBNSoftware** [comment]
Ok. I understand.

Thank you !
